src/index.ts

```diff
@@ -1,164 +1,633 @@
+  components: Components
+  confidences: {
+    collect: () => Promise<Maybe<string>> // returns username
+    request: () => Promise<void>
+    session: (username: string) => Promise<Maybe<Session>>
+  }
+  session: Maybe<Session>
+}
+
+
+export enum ProgramError {
+  InsecureContext = "INSECURE_CONTEXT",
+  UnsupportedBrowser = "UNSUPPORTED_BROWSER"
+}
+
 
-import { impl as auth } from "./auth/implementation.js"
-import { isPermissionedAppState } from "./auth/state.js"
-import { setImplementations } from "./setup.js"
-import { BASE_IMPLEMENTATION } from "./auth/implementation/base.js"
-import { USE_WNFS_IMPLEMENTATION } from "./auth/implementation/use-wnfs.js"
-import { LOBBY_IMPLEMENTATION } from "./auth/implementation/lobby.js"
-import { AppState } from "./auth/state/app.js"
-import { PermissionedAppState } from "./auth/state/permissionedApp.js"
-import * as appState from "./auth/state/app.js"
-import * as permissionedAppState from "./auth/state/permissionedApp.js"
+export type ShortHands = {
+  loadFileSystem: (username: string) => Promise<FileSystem>
+  loadRootFileSystem: (username: string) => Promise<FileSystem>
+}
+
+
+export type AuthenticationStrategies = Record<
+  string,
+  {
+    implementation: Auth.Implementation<Components>
+
+    accountConsumer: (username: string) => Promise<AccountLinkingConsumer>
+    accountProducer: (username: string) => Promise<AccountLinkingProducer>
+    isUsernameAvailable: (username: string) => Promise<boolean>
+    isUsernameValid: (username: string) => Promise<boolean>
+    register: (options: { username: string; email?: string }) => Promise<{ success: boolean }>
+    session: () => Promise<Maybe<Session>>
+  }
+>
+
+
+export const strategyTypes = {
+  default: BaseAuth.TYPE,
+  webCrypto: BaseAuth.TYPE
+}
+
+
+
+// ENTRY POINTS
 
 
 /**
- * Check if we're authenticated and initiate the user's file system if 
- * authenticated (can be disabled).
+ * Build a webnative program.
  *
- * See `loadFileSystem` if you want to load the user's file system yourself.
- * NOTE: Only works on the main/ui thread, as it uses `window.location`.
+ * This will give you a `Program` object which has the following properties:
+ * - `session`, a `Session` object if a session was created before.
+ * - `auth`, a means to control the various auth strategies you configured. Use this to create sessions. Read more about auth components in the toplevel `auth` object documention.
+ * - `confidences`, a means to control confidences. Use this to collect & request confidences, and to create a session based on them. Read more about confidences in the toplevel `confidences` object documentation.
+ * - `components`, your full set of `Components`.
+ *
+ * This object also has a few other functions, eg. to load a filesystem.
+ * These are called "shorthands" because they're the same functions available
+ * through other places in webnative, but you don't have to pass in the components.
  *
+ * See `assemble` for more information.
```

Comment:
Can we somehow make it so that typedoc inserts a link here to the documentation of this `assemble` function?